lldb/include/lldb/Symbol/SwiftASTContext.h

```diff
@@ -134,28 +137,28 @@ class TypeSystemSwift : public TypeSystem {
   virtual lldb::TypeSP GetCachedType(ConstString mangled) = 0;
   virtual void SetCachedType(ConstString mangled,
                              const lldb::TypeSP &type_sp) = 0;
-  virtual bool IsImportedType(CompilerType type,
+  virtual bool IsImportedType(lldb::opaque_compiler_type_t type,
```

Comment:
I'd imagine that here, since `IsImportedType` should only be used/overriden inside of the Swift plugin, it'd be safe to substitute `swift::TypeBase *` for `opaque_compiler_type_t`?

I see that we generally try not to #include swift headers in SwiftASTContext.h, since that header gets pulled into all different kinds of places like source/Core and source/Expression. But we could forward-declare `swift::TypeBase` in SwiftASTContext.h and probably get away with using it that way?

Reply:
I see what you mean now. I'm primarily aiming at consistency here. The idea is that these functions be threaded through CompilerType and CompilerType doing the dynamic dispatch based on the TypeSystem in the CompilerType. However, since I didn't actually thread this through CompilerType I admit that this looks weird.

But, it allows us to write code like this: 

```
imported_self_type =
       llvm::cast<TypeSystemSwift>(imported_self_type.GetTypeSystem())
           ->GetReferentType(imported_self_type);
           ->GetReferentType(imported_self_type.GetOpaqueQualType());
```

that works with both a TypeSystemSwiftTypeRef and a SwiftASTContext CompilerType. If we do the CompilerType extension, this can become:

```
imported_self_type = imported_self_type.GetReferentType();
```

It's just that GetReferentType is such an odd duck of a function that I don't want to bake it into CompilerType now.